launcher/sdw_updater_gui/Updater.py

```diff
@@ -91,86 +63,29 @@ def apply_updates(vms):
-    """
-    Check for dom0 updates
-    """
-    try:
-        subprocess.check_call(["sudo", "qubes-dom0-update", "--check-only"])
-    except subprocess.CalledProcessError as e:
-        sdlog.error("dom0 updates required or cannot check for updates")
-        sdlog.error(str(e))
-        return UpdateStatus.UPDATES_REQUIRED
-
-    sdlog.info("dom0 is up to date")
-    return UpdateStatus.UPDATES_OK
-
-
-def _check_updates_fedora():
-    """
-    Check for updates to the default Fedora TemplateVM. Fedora has a very rapid
-    release cycle and there are almost always updates to fedora VMs. Let's just
-    return UPDATES_REQUIRED and always upgrade those VMs, since they no longer
-    trigger a full workstation reboot on upgrade.
-    """
-    return UpdateStatus.UPDATES_REQUIRED
-
-
-def _check_updates_debian(vm):
-    """
-    Check for updates for a given Debian-based TemplateVM
-    """
-    updates_required = False
-    try:
-        # There is no apt command that uses exit codes in such a way that we can discover if
-        # updates are required by relying on exit codes.
-        # Since we don't want to use --pass-io and parse the output, we have to count
-        # the lines on the vm output
-        sdlog.info("Checking for updates {}:{}".format(vm, current_templates[vm]))
-        subprocess.check_call(["qvm-run", current_templates[vm], "sudo apt update"])
-        subprocess.check_call(
-            [
-                "qvm-run",
-                current_templates[vm],
-                "[[ $(apt list --upgradable | wc -l) -eq 1 ]]",
-            ]
-        )
-    except subprocess.CalledProcessError as e:
-        sdlog.error(
-            "Updates required for {} or cannot check for updates".format(
-                current_templates[vm]
-            )
-        )
-        sdlog.error(str(e))
-        updates_required = True
-    finally:
-        reboot_status = _safely_shutdown_vm(current_templates[vm])
-        if reboot_status == UpdateStatus.UPDATES_FAILED:
-            return reboot_status
-
-    if not updates_required:
-        sdlog.info("{} is up to date".format(current_templates[vm]))
-        return UpdateStatus.UPDATES_OK
-    else:
-        return UpdateStatus.UPDATES_REQUIRED
-
-
 def _apply_updates_dom0():
     """
     Apply updates to dom0. Any update to dom0 will require a reboot after
     the upgrade.
     """
     sdlog.info("Updating dom0")
     try:
-        subprocess.check_call(["sudo", "qubes-dom0-update", "-y"])
+        output = subprocess.check_output(["sudo", "qubes-dom0-update", "-y"]).decode(
+            "utf-8"
+        )
     except subprocess.CalledProcessError as e:
         sdlog.error(
             "An error has occurred updating dom0. Please contact your administrator."
         )
         sdlog.error(str(e))
         return UpdateStatus.UPDATES_FAILED
-    sdlog.info("dom0 update successful")
-    return UpdateStatus.REBOOT_REQUIRED
+
+    if output.find("No packages downloaded") != -1:
```

Comment:
@emkll This is the most significant business logic change, I think. The return code from qubes-dom0-update does not appear to tell us reliably whether or not updates were applied, and we need to know in order to trigger the reboot logic. So I decided to look at the command output. Based on my review of the updater logic in `/usr/lib/qubes/qubes-download-dom0-updates.sh` in `sys-firewall`, my understanding is that this string should be present whenever a `dom0` update was successful but yielded no results.

Reply:
@conorsch suggested maybe running the `--check-only` stage here after all so we can use its return code to determine the reboot requirement. In `time` runs that only adds a few seconds as the index is now sufficiently up-to-date for it to use it on the subsequent run. I'll leave it as is for now for your initial review, let me know what you think, and if you have other ideas for how to handle this case.

Reply:
OK, I'll switch to using a `--check-only` run for `dom0`.

Reply:
This is done in 0e5d1724b16eb29c7ee469e532a0f8055cb62ae2.